docfx_yaml/extension.py

```diff
@@ -1527,6 +1533,78 @@ def search_cross_references(obj, current_object_name: str, known_uids: List[str]
                     markdown_utils.reformat_markdown_to_html(attribute_type))
 
 
+# Type alias used for toc_yaml entries.
+_toc_yaml_type_alias = dict[str, any]
+
+def merge_markdown_and_package_toc(
+    app: sphinx.application,
+    pkg_toc_yaml: list[_toc_yaml_type_alias],
+    markdown_toc_yaml: _toc_yaml_type_alias,
+    known_uids: set[str],
+) -> tuple[MutableSet[str], list[_toc_yaml_type_alias]]:
+    """
+    Merges the markdown and package table of contents.
+
+    Args:
+        app: Sphinx application.
+        pkg_toc_yaml: table of content for package files.
+        markdown_toc_yaml: table fo content for markdown files.
+
+    Returns:
+        A set of markdown pages that has been added, and the merged table of
+        contents file, with files in the correct position.
+    """
+    def _flatten_toc(
+        toc_yaml_entry: list[_toc_yaml_type_alias],
+    ) -> list[_toc_yaml_type_alias]:
+        """Flattens and retrieves all children within pkg_toc_yaml."""
+        entries = list(toc_yaml_entry)
+        for entry in toc_yaml_entry:
+            if (children := entry.get('items')):
+                entries.extend(_flatten_toc(children))
+                entries.extend(children)
```

Comment:
Does this line end up double-adding? `_flatten_toc` is called with `children`, which will then return `children` (and more), then we add `children` again right here.

Seems like this needs a test.

Reply:
🤦🤦 Yes you're right. Removed the redundant line, added unit test coverage as well. I didn't add it before because it required `sphinx.app` ... before, but not anymore so fine to add now.